collections/docs/01-intro.md

```diff
@@ -0,0 +1,232 @@
+Which makes collections dumb, but also means that hypothetically I can store everything
+that can be a go type into a collection. I am not bounded to any type of encoding (be it proto, json or whatever)
+
+So a collection needs to be given a way to understand how to convert your keys and values to bytes. 
+This is achieved through ``KeyCodec`` and `ValueCodec`, which are arguments that you pass to your 
+collections when you're instantiating them using the ```collections.NewMap/collections.NewItem/...``` 
+instantiation functions.
+
+(NOTE: generally speaking you will never be required to implement your own ``Key/ValueCodec`` as 
+the SDK and collections libraries already come with default, safe and fast implementation of those.
+You might need to implement them only if you're migrating to collections and there are state layout incompatibilities.)
+
+Let's make an example:
+
+````go
+package collections
+
+import (
+	"cosmossdk.io/collections"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var IDsPrefix = collections.NewPrefix(0)
+
+type Keeper struct {
+	Schema    collections.Schema
+	IDs   collections.Map[string, uint64]
+}
+
+func NewKeeper(storeKey *storetypes.KVStoreKey) Keeper {
+	sb := collections.NewSchemaBuilder(sdk.OpenKVStore(storeKey))
+
+	return Keeper{
+		IDs: collections.NewMap(sb, IDsPrefix, "ids", collections.StringKey, collections.Uint64Value),
+	}
+}
+````
+
+We're now instantiating a map where the key is string and the value is uint64.
+We already know the first three arguments of the ``NewMap`` function.
+
+The fourth parameter is our `KeyCodec`, we know that the ``Map`` has `string` as key so we pass it a `KeyCodec` that handles strings as keys.
+
+The fifth parameter is our `ValueCodec`, we know that the `Map` as a `uint64` as value so we pass it a `ValueCodec` that handles uint64.
+
+Collections already comes with all the required implementations for golang primitive types.
+
+Let's make another example, this falls closer to what we build using cosmos SDK, let's say I want
+to create a `collections.Map` that maps account addresses to their base account. So we want to map an `sdk.AccAddress` to an `auth.BaseAccount` (which is a proto):
+
+```go
+package collections
+
+import (
+	"cosmossdk.io/collections"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+var AccountsPrefix = collections.NewPrefix(0)
+
+type Keeper struct {
+	Schema    collections.Schema
+	Accounts   collections.Map[sdk.AccAddress, authtypes.BaseAccount]
+}
+
+func NewKeeper(storeKey *storetypes.KVStoreKey, cdc codec.BinaryCodec) Keeper {
+	sb := collections.NewSchemaBuilder(sdk.OpenKVStore(storeKey))
+	return Keeper{
+		Accounts: collections.NewMap(sb, AccountsPrefix, "accounts",
+			sdk.AccAddressKey, codec.CollValue[authtypes.BaseAccount](cdc)),
+	}
```

Comment:
I wonder if it would make it easier for devs/clients if instead `collections.New*` took a struct literal, so you know what each param means.

e.g.

```go
collections.New(
  collections.Config{
    SchemaBuilder: db,
    Prefix: AccountsPrefix,
    // ...
  }
)
```